### PR TITLE
Support for simple cache

### PR DESCRIPTION
**Pull request by Nyholm**
Introducing the support for cache proxies instead for messing around with different types of recording pools like we did. 

This will create a proxy file that extends the original pool. It is only used in development of course.

### CODE REVIEW

File: src/DataCollector/ProxyFactory.php [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+    }
+
+    private function checkProxyDirectory()
+    {
+        if (!is_dir($this->proxyDirectory)) {
+            @mkdir($this->proxyDirectory, 0777, true);
+        }
+    }
+
+    protected function getProxyClass($namespace)
```
Comment (Nyholm) [inline src/DataCollector/ProxyFactory.php]: Should be private

File: src/DataCollector/DecoratingFactory.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+     * @param ProxyFactory $proxyFactory
+     * @param mixed        $originalObject
+     */
+    public function __construct(ProxyFactory $proxyFactory, $originalObject)
+    {
+        $this->proxyFactory   = $proxyFactory;
+        $this->originalObject = $originalObject;
+    }
+
+    public function create()
```
Comment (Nyholm) [inline src/DataCollector/DecoratingFactory.php]: Use $originalObject as a function parameter and register the factory in the service.yml

File: src/DataCollector/ProxyFactory.php [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+    public function __construct($proxyDirectory)
+    {
+        $this->proxyDirectory = $proxyDirectory;
+    }
+
+    /**
+     * Create a proxy that handles logging better.
+     *
+     * @param string $class
+     * @param string $file  where we store the proxy class
```
Comment (Nyholm) [inline src/DataCollector/ProxyFactory.php]: Wrong variable name

File: src/DataCollector/ProxyFactory.php [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+    {
+        $proxyClass = $this->getProxyClass($class);
+        $class      = '\\'.rtrim($class, '\\');
+        $proxyFile  = $this->proxyDirectory.'/'.$proxyClass.'.php';
+
+        if (class_exists($proxyClass)) {
+            return $proxyClass;
+        }
+
+        $content = <<<PROXY
```
Comment (Nyholm) [inline src/DataCollector/ProxyFactory.php]: Could we `get_file_contents()` this file instead? It will be easier to change and review.

**Review by Nyholm — COMMENTED** [review]
What happened with logging support?

**Comment by prisis** [comment]
sorry had no time for a review -.-

**Comment by Nyholm** [comment]
No worries. I have 2 PRs open now. After that I think we should release a new version. What do you think?

File: composer.json [hunk 5]
```diff
@@ -27,14 +27,15 @@
         "php": "^5.6 || ^7.0",
         "symfony/framework-bundle": "^2.7 || ^3.0",
         "cache/taggable-cache": "^0.5",
-        "cache/session-handler": "^0.2"
+        "cache/session-handler": "^0.2",
```
Comment (prisis) [inline composer.json:30]: we have stable version, why dont use them?

Comment (Nyholm) [reply composer.json:30]: It was not a part of this PR. 

But here: https://github.com/php-cache/cache-bundle/pull/76

**Comment by prisis** [comment]
Sounds like a good idea

**Pull request merged**